### ISSUE TITLE
commit message regex does not work as Python/Java regex.

### ISSUE THREAD

**Issue by luozhaohui** [mohamicorp/yet-another-commit-checker#234]
I try to reject commit message which has Jira ID only  with the follow regex expression:
`\s*\S+\s*[, :-]+[A-Z][A-Z_0-9]*-[0-9]+.*|.*[A-Z][A-Z_0-9]*-[0-9]+[, :-]+\s*\S+\s*`

I tested it with the following cases in Python and it worked.
```
        'TASK-8156: review adaptions',
        'JIRA ID: TASK-8156: review adaptions',
```

It also worked in Java as:
```
Pattern p = Pattern.compile("(?:\\s*\\S+\\s*[, :-]+[A-Z][A-Z_0-9]*-[0-9]+.*|.*[A-Z][A-Z_0-9]*-[0-9]+[, :-]+\\s*\\S+\\s*)");
```

But yacc rejected these messages. Could you look into this issue?

**Comment by eddyp** [comment]
YACC does not match on the entire commit message, currently, which also bit me, too.
See #100 #144

**Comment by george-vulov** [comment]
Fix tracked on #207 -- note that there is a workaround noted on the ticket, prefix your regex with `(?s)`.
